main.go

```diff
@@ -118,18 +122,25 @@ func main() {
 	flag.Var(&commonRunnerLabels, "common-runner-labels", "Runner labels in the K1=V1,K2=V2,... format that are inherited all the runners created by the controller. See https://github.com/actions-runner-controller/actions-runner-controller/issues/321 for more information")
 	flag.StringVar(&namespace, "watch-namespace", "", "The namespace to watch for custom resources. Set to empty for letting it watch for all namespaces.")
 	flag.StringVar(&logLevel, "log-level", logging.LogLevelDebug, `The verbosity of the logging. Valid values are "debug", "info", "warn", "error". Defaults to "debug".`)
+	flag.BoolVar(&enableProdLogConfig, "enable-prod-log-config", false, "use Zap production config which will default to json format")
+
 	flag.Parse()
+	
+	if enableProdLogConfig {
+		logging.LogOpts.Development = false
+		logging.LogOpts.TimeEncoder = nil 
+	} 
```

Comment:
Can we implement an additional logger factory function on top of `logging.NewLogger` that takes the log format string `text` or `json` so that we can abstract this if-block away?